### ISSUE TITLE
Changing highway=construction, construction=residential to residential leaves extra construction tag

### ISSUE THREAD

**Issue by matthewdarwin** [openstreetmap/iD#5472]
Yesterday I edited https://www.openstreetmap.org/way/569768740

I used the nice dropdown list id iD to change highway from construction to residential.  However, in the same edit, iD failed to remove the construction=residential tag.... and it wasn't shown to me to manually remove either.   

I saw the same behaviour (broken tags) on highway=proposed, proposed=residential on an edit by someone else on the previous day.

So seems iD understands the relationship between highway=x, x=residential (for x=proposed, construction), but fails to make proper updates.

Possibly related: #4659 #2021

**Comment by matkoniecz** [comment]
> iD presets won't be able to handle all the different kinds of lifecycle tagging that exist in the wild.

Obviously, but `highway=*` is standard one and is widely used - so hopefully it will be supported someday.

On more case that prompted me to report this bug, starting from search for duplicates: https://www.openstreetmap.org/way/214320767/history
